claim/claimstore.go

```diff
@@ -52,22 +53,24 @@ func (s Store) Read(name string) (Claim, error) {
 	return claim, err
 }
 
-// ReadAll retrieves all the claims
+// ReadAll retrieves all of the claims.
 func (s Store) ReadAll() ([]Claim, error) {
-	claims := make([]Claim, 0)
-
-	list, err := s.backingStore.List()
+	results, err := s.backingStore.ReadAll()
 	if err != nil {
-		return claims, err
+		return nil, err
 	}
 
-	for _, c := range list {
-		cl, err := s.Read(c)
+	claims := make([]Claim, len(results))
+	for i := range claims {
```

Comment:
Totally nit, but what about ranging on results instead of claims?
```go
for i, bytes := range results{
    var claim Claim
    if err := json.Unmarshal(bytes, &claim); err != nil {
        return nil, fmt.Errorf("error unmarshaling claim: %v", err)
    }
    claims[i] = claim
}
```